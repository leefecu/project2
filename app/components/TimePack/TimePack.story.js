import React from 'react';
import { Text, View } from 'react-native';

import { storiesOf } from '@storybook/react-native';
import { action } from '@storybook/addon-actions';

import TimePack from './TimePack';
import Styles from '../../Stories.styles';

storiesOf('Component | TimePack', module).add('Default', () => {
  return (
    <View style={Styles.defaultBGContainer}>
      <View style={Styles.row}>
        <Text style={Styles.title}>15 minutes</Text>
        <TimePack
          duration={15}
          packName="15 minutes"
          onPress={action('onPress')}
          packetId="id1"
          price={0.4}
        />
      </View>
      <View style={Styles.row}>
        <Text style={Styles.title}>1 hour</Text>
        <TimePack
          duration={60}
          packName="1 hour"
          onPress={action('onPress')}
          packetId="id2"
          price={1}
        />
      </View>
      <View style={Styles.row}>
        <Text style={Styles.title}>with daily message</Text>
        <TimePack
          duration={180}
          packName="3 hours"
          onPress={action('onPress')}
          packetId="id2"
          price={5}
          messageBody="This is a testing packet level notification."
          messageTitle="Testing PACKET Notification"
        />
      </View>
      <View style={Styles.row}>
        <Text style={Styles.title}>disabled</Text>
        <TimePack
          duration={1200}
          packName="20 hours"
          onPress={action('onPress')}
          packetId="id3"
          price={1.5}
          disabled
        />
      </View>
    </View>
  );
});
