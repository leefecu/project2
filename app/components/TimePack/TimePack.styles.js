import { StyleSheet } from 'react-native';

import Colors from '../../themes/Colors';
import Metrics from '../../themes/Metrics';

export const TIME_PACK_WIDTH = Metrics.width - Metrics.doublePadding * 2;

export default StyleSheet.create({
  container: {
    alignItems: 'stretch',
    flexDirection: 'row',
    width: TIME_PACK_WIDTH,
  },
  messageBG: {
    backgroundColor: Colors.primaryLight,
  },
  labelColumn: {
    alignItems: 'flex-start',
    backgroundColor: Colors.white,
    borderTopLeftRadius: Metrics.inputBorderRadius,
    borderBottomLeftRadius: Metrics.inputBorderRadius,
    flex: 6,
    justifyContent: 'center',
    paddingHorizontal: Metrics.basePadding,
    paddingVertical: Metrics.basePadding,
  },
  disabledLabelColumn: {
    backgroundColor: Colors.lightGrey,
  },
  buttonColumn: {
    alignItems: 'center',
    backgroundColor: Colors.primary,
    borderTopRightRadius: Metrics.inputBorderRadius,
    borderBottomRightRadius: Metrics.inputBorderRadius,
    flex: 4,
    justifyContent: 'center',
    paddingVertical: Metrics.smallPadding,
  },
  disabledButtonColumn: {
    backgroundColor: Colors.stoneGrey,
  },
  packNameBlock: {
    alignItems: 'flex-end',
  },
  packName: {
    color: Colors.primaryDark,
    fontSize: Metrics.Fonts.small,
    fontFamily: 'CircularPro-Bold',
    marginRight: Metrics.smallestMargin,
  },
  unit: {
    color: Colors.primaryDark,
    fontSize: Metrics.Fonts.small,
    fontFamily: 'CircularPro-Bold',
  },
  disabledText: {
    color: Colors.disabledText,
  },
  topupLabel: {
    fontSize: Metrics.Fonts.smallest,
    fontFamily: 'CircularPro-Book',
  },
  message: {
    borderTopColor: Colors.lineColor,
    borderTopWidth: 1,
    marginTop: Metrics.basePadding,
    paddingTop: Metrics.basePadding,
  },
  messageTitle: {
    color: Colors.primaryDark,
    fontSize: Metrics.Fonts.smaller,
    fontFamily: 'CircularPro-Bold',
    marginBottom: Metrics.baseMargin,
  },
  messageBody: {
    color: Colors.descriptionColor,
    fontSize: Metrics.Fonts.smallest,
    fontFamily: 'CircularPro-Book',
  },
});
