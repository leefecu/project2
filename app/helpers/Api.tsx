import get from 'lodash/get';
import reject from 'lodash/reject';
import Config from 'react-native-config';
import compareVersions from 'compare-versions';
import { Platform } from 'react-native';

import * as Storage from './Storage';
import { createExpiryNotification } from './Notification';
import { convertStashToActivePAck } from './Transformation';
import API from '../services/Api';

import { SYSTEM_RESPONSE } from '../constants/Shared';

const api = API.create();

export const purchasePack = async (packetId: string, promoId?: string) => {
  const response = await api.purchasePack(packetId, promoId);
  const currentActivePack = await Storage.getActivePack();
  if (response.ok) {
    const balance = get(response, 'data.balance');
    const purchasedPack = get(response, 'data.purchasedPack');
    const activePack = currentActivePack
      ? {
          ...currentActivePack,
          endTime: purchasedPack.endTime,
        }
      : purchasedPack;

    createExpiryNotification(purchasedPack.endTime);
    await Storage.setActivePackAndBalance(activePack, balance);
    return response;
  }
  return response;
};

export const activateStash = async (stash: TStash) => {
  const response = await api.activateStash(stash.packetInstanceId);
  if (response.ok) {
    const currentActivePack = await Storage.getActivePack();
    const currentStash = await Storage.getStash();
    const newActivePack = convertStashToActivePAck(stash, currentActivePack);
    const newStash = reject(
      currentStash,
      (target: TStash) => target.packetInstanceId === stash.packetInstanceId,
    );

    createExpiryNotification(newActivePack.endTime);
    await Storage.saveActivePackAndStash(newActivePack, newStash);
    return response;
  }
  return response;
};

export const checkSystemInfo = async (): Promise<SYSTEM_RESPONSE> => {
  const response = await api.getSystemInfo();
  if (response.ok) {
    const maintenance: boolean = get(response, 'data.maintenance', false);
    if (maintenance) {
      return SYSTEM_RESPONSE.MAINTENANCE;
    }
    const minVersion = get(response, `data.${Platform.OS}`);
    const appVersion = Config.APP_VERSION;

    return compareVersions.compare(appVersion, minVersion, '>=')
      ? SYSTEM_RESPONSE.CONTINUE
      : SYSTEM_RESPONSE.UPGRADE;
  }
  return SYSTEM_RESPONSE.CONTINUE;
};
