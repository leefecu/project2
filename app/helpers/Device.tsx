import { NativeModules, Platform } from 'react-native';
import DeviceInfo from 'react-native-device-info';
import CarrierInfo from 'react-native-carrier-info';
import RNFS from 'react-native-fs';

const NZ_MCC = '530';
const NZ_COUNTRY_CODE = 'nz';
const IS_IOS = Platform.OS === 'ios';

const NetworkInfo = NativeModules.RNNetworkInfo;

export const isSimulator = async (): Promise<boolean> => {
  const isEmulator = await DeviceInfo.isEmulator();
  return isEmulator;
};

export const isSimIn = async () => {
  // react-native-carrier-info does not work in simulator
  if (await isSimulator()) {
    return true;
  }

  try {
    // mobileCountryCode method fails when no SIM card in the device
    await CarrierInfo.mobileCountryCode();
    return true;
  } catch (e) {
    return false;
  }
};

export const checkForRoaming = async (debug: boolean = false): Promise<boolean> => {
  // react-native-carrier-info does not work in simulator
  if (await isSimulator()) {
    return debug;
  }

  try {
    if (Platform.OS === 'ios') {
      const operatorPlist = await NativeModules.NetworkInfo.getOperatorPlist();
      const mcc = operatorPlist.match(/\d+\/carrier.plist/g);
      if (mcc && mcc.length > 0) {
        return mcc[0].substr(0, 3) !== NZ_MCC;
      }
      return false;
    }
    const countryCode = await NetworkInfo.isoCountryCode();
    return countryCode !== NZ_COUNTRY_CODE;
  } catch (e) {
    return false;
  }
};

const majorVersionIOS: number = parseInt(`${Platform.Version}`, 10);

export const isAutoFillSupported: boolean = IS_IOS && majorVersionIOS >= 12;

// Figured out how current(old) DataClock app stores msisdn
// it's stored in xml file
// shared_prefs/nz.co.twodegreesmobile.comptelfwd.smpl.Account.xml
const msisdnFileName = 'nz.co.twodegreesmobile.comptelfwd.smpl.Account.xml';
const msisdnFolderName = 'shared_prefs';
const msisdnRegex = /com.comptel.smpl.interaction.Account.PHONE_NUMBER_KEY">\d+<\/string/g;

export const getExistingMsisdn = async (): Promise<TMigrationData> => {
  if (IS_IOS) {
    try {
      const msisdn = await NativeModules.GetNumber.getMsisdn();
      return { number: msisdn, upgrade: !!msisdn };
    } catch (e) {
      return { number: '', upgrade: false };
    }
  }
  // This method only works with Android
  try {
    // e.g. RNFS.DocumentDirectoryPath = '/data/user/0/nz.co.twodegreesmobile.comptelfwd.smpl/files',
    const paths = RNFS.DocumentDirectoryPath.split('/');
    // need to remove "/files" and replace with "shared_prefs"
    paths[paths.length - 1] = msisdnFolderName;
    const path = paths.join('/');
    const xmlPath = `${path}/${msisdnFileName}`;
    const existingAccount = await RNFS.readFile(xmlPath);
    const match = msisdnRegex.exec(existingAccount);
    if (match && match.length > 0) {
      const msisdn = match[0]
        .replace('com.comptel.smpl.interaction.Account.PHONE_NUMBER_KEY">', '')
        .replace('</string', '');

      return { number: msisdn, upgrade: true };
    }
    return { number: '', upgrade: false };
  } catch (e) {
    return { number: '', upgrade: false };
  }
};
