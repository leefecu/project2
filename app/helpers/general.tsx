import { NavigationParams, NavigationScreenProp, NavigationState } from 'react-navigation';

import { ERROR_CODE } from '../constants/Messages';
import { SYSTEM_RESPONSE } from '../constants/Shared';

import * as API from '../helpers/Api';

export const noop = () => {};

interface TTestProp {
  testID: string;
  accessibilityLabel: string;
}
export const testProps = (id: string): TTestProp => ({
  testID: id,
  accessibilityLabel: id,
});

export const shouldLogout = (status: number | undefined, errorCode: ERROR_CODE): boolean =>
  status === 401 ||
  status === 403 ||
  errorCode === ERROR_CODE.NOT_REGISTERED ||
  errorCode === ERROR_CODE.INVALID_PLAN ||
  errorCode === ERROR_CODE.QD_ISSUE;

export const systemCheck = async (
  navigation: NavigationScreenProp<NavigationState, NavigationParams>,
) => {
  const systemInfo = await API.checkSystemInfo();
  if (systemInfo === SYSTEM_RESPONSE.CONTINUE) {
    return;
  }
  navigation.navigate(systemInfo === SYSTEM_RESPONSE.MAINTENANCE ? 'Maintenance' : 'Upgrade');
};
