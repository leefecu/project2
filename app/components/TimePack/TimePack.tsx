import React from 'react';
import { Text, TouchableHighlight, View } from 'react-native';

import Button from '../Button/Button';

import { testProps } from '../../helpers/general';

import Styles, { TIME_PACK_WIDTH } from './TimePack.styles';
import Colors from '../../themes/Colors';
import Metrics from '../../themes/Metrics';

interface Props extends TTimePack {
  onPress: (packetId: string) => void;
}

const BUTTON_WIDTH = TIME_PACK_WIDTH * 0.4 - Metrics.basePadding * 2;

const parsePrice = (price: number): string => (price === 0 ? 'Free' : `$${price.toFixed(2)}`);

const TimePack = ({
  disabled,
  messageBody,
  messageTitle,
  onPress,
  packName,
  packetId,
  price,
}: Props) => {
  const handlePress = () => {
    onPress(packetId);
  };
  return (
    <TouchableHighlight
      activeOpacity={disabled ? 1 : 0.5}
      onPress={handlePress}
      underlayColor={Colors.transparent}
      {...testProps(`Start Time Pack ${packName}`)}>
      <View style={Styles.container}>
        <View style={[Styles.labelColumn, disabled && Styles.disabledLabelColumn]}>
          <View style={Styles.packNameBlock}>
            <Text style={[Styles.packName, disabled && Styles.disabledText]}>{packName}</Text>
          </View>
          {disabled && <Text style={[Styles.disabledText, Styles.topupLabel]}>Top up to buy</Text>}
          {messageTitle !== '' && !disabled && (
            <View style={Styles.message}>
              <Text style={Styles.messageTitle}>{messageTitle}</Text>
              {messageBody && <Text style={Styles.messageBody}>{messageBody}</Text>}
            </View>
          )}
        </View>
        <View
          style={[
            Styles.buttonColumn,
            !!messageTitle && Styles.messageBG,
            disabled && Styles.disabledButtonColumn,
          ]}>
          <Button
            disabled={disabled}
            disregardDisablePress={true}
            onPress={handlePress}
            id={`Purchase Pack ${packName}`}
            label={parsePrice(price)}
            smallButton
            width={BUTTON_WIDTH}
            withBG={!disabled}
          />
        </View>
      </View>
    </TouchableHighlight>
  );
};

TimePack.defaultProps = {
  disabled: false,
};

export default TimePack;
