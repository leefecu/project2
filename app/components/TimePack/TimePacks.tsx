import React, { PureComponent } from 'react';
import { FlatList, View } from 'react-native';
import throttle from 'lodash/throttle';

import TimePack from './TimePack';

import { THROTTLING } from '../../constants/Shared';

import { noop } from '../../helpers/general';

import Styles from './TimePacks.styles';

interface Props {
  balance: number;
  disableOnBalance: boolean;
  onSelectPack: (pack: TTimePack, lowBalance: boolean) => void;
  packs: Array<TTimePack>;
}

class TimePacks extends PureComponent<Props> {
  static defaultProps: Props = {
    balance: 0,
    disableOnBalance: false,
    onSelectPack: noop,
    packs: [],
  };

  handleSelectPackThrottled: Function;

  constructor(props: Props) {
    super(props);

    this.handleSelectPackThrottled = throttle(this.handleSelectPack, THROTTLING, {
      trailing: false,
    });
  }

  disabled = (pack: TTimePack): boolean => pack.disabled || this.lowBalance(pack);

  handleSelectPack = (item: TTimePack, lowBalance: boolean) => {
    const { onSelectPack } = this.props;
    onSelectPack(item, lowBalance);
  };

  lowBalance = (pack: TTimePack): boolean => {
    const { balance, disableOnBalance } = this.props;
    return disableOnBalance && balance < pack.price;
  };

  keyExtractor = (item: TTimePack) => item.packetId;

  renderPack = ({ item }: { item: TTimePack }) => {
    return (
      <View style={Styles.packRow}>
        <TimePack
          onPress={() => this.handleSelectPackThrottled(item, this.lowBalance(item))}
          {...item}
          disabled={this.disabled(item)}
        />
      </View>
    );
  };

  render() {
    const { packs } = this.props;
    return <FlatList data={packs} keyExtractor={this.keyExtractor} renderItem={this.renderPack} />;
  }
}

export default TimePacks;
