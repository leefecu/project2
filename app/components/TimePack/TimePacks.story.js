import React from 'react';
import { View } from 'react-native';

import { storiesOf } from '@storybook/react-native';

import TimePacks from './TimePacks';
import Styles from '../../Stories.styles';

import { PACKS } from '../../__testData__/offers';

storiesOf('Component | TimePacks', module).add('Default', () => {
  return (
    <View style={Styles.defaultBGContainer}>
      <View style={Styles.row}>
        <TimePacks packs={PACKS} />
      </View>
    </View>
  );
});
