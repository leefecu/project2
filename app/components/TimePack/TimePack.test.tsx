/**
 * @jest-environment jsdom
 */
import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';

import TimePack from './TimePack';

import { noop } from '../../helpers/general';

describe('TimePack component', () => {
  test('render default', () => {
    const button = create(
      <TimePack duration={15} packName="15 minutes" onPress={noop} packetId="id1" price={0.4} />,
    );
    expect(button.toJSON()).toMatchSnapshot();
  });

  test('render with message', () => {
    const button = create(
      <TimePack
        duration={180}
        packName="3 hours"
        onPress={noop}
        packetId="id2"
        price={5}
        messageBody="This is a testing packet level notification."
        messageTitle="Testing PACKET Notification"
      />,
    );
    expect(button.toJSON()).toMatchSnapshot();
  });

  test('trigger onPress function upon press', () => {
    const onPressEvent = jest.fn();
    const component = shallow(
      <TimePack
        duration={15}
        packName="15 minutes"
        onPress={onPressEvent}
        packetId="id1"
        price={0.4}
      />,
    );
    component.find('Button').simulate('press');
    expect(onPressEvent).toHaveBeenCalled();
  });
});
