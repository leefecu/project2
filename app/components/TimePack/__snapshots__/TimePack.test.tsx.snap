// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimePack component render default 1`] = `
<View
  accessibilityLabel="Start Time Pack 15 minutes"
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
  testID="Start Time Pack 15 minutes"
>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "flexDirection": "row",
        "width": 702,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "backgroundColor": "#ffffff",
            "borderBottomLeftRadius": 5,
            "borderTopLeftRadius": 5,
            "flex": 6,
            "justifyContent": "center",
            "paddingHorizontal": 16,
            "paddingVertical": 16,
          },
          false,
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-end",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#004071",
                "fontFamily": "CircularPro-Bold",
                "fontSize": 17,
                "marginRight": 2,
              },
              false,
            ]
          }
        >
          15 minutes
        </Text>
      </View>
      <View
        style={
          Object {
            "borderTopColor": "#979797",
            "borderTopWidth": 1,
            "marginTop": 16,
            "paddingTop": 16,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#004071",
              "fontFamily": "CircularPro-Bold",
              "fontSize": 15,
              "marginBottom": 16,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#00a7e5",
            "borderBottomRightRadius": 5,
            "borderTopRightRadius": 5,
            "flex": 4,
            "justifyContent": "center",
            "paddingVertical": 12,
          },
          false,
          false,
        ]
      }
    >
      <View
        accessibilityLabel="Purchase Pack 15 minutes button"
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "width": 248.8,
            },
            Object {
              "alignItems": "stretch",
              "backgroundColor": "transparent",
              "borderColor": "#004071",
              "borderRadius": 32,
              "borderWidth": 1,
              "paddingVertical": 12,
            },
            Object {
              "paddingVertical": 8,
            },
            Object {
              "borderColor": "rgba(255, 255, 255, 0.7)",
            },
            false,
            false,
            false,
          ]
        }
        testID="Purchase Pack 15 minutes button"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#004071",
                "fontFamily": "CircularPro-Bold",
                "fontSize": 20,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              Object {
                "fontSize": 15,
              },
              Object {
                "color": "#ffffff",
              },
              false,
              false,
              false,
            ]
          }
        >
          $0.40
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`TimePack component render with message 1`] = `
<View
  accessibilityLabel="Start Time Pack 3 hours"
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
  testID="Start Time Pack 3 hours"
>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "flexDirection": "row",
        "width": 702,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "backgroundColor": "#ffffff",
            "borderBottomLeftRadius": 5,
            "borderTopLeftRadius": 5,
            "flex": 6,
            "justifyContent": "center",
            "paddingHorizontal": 16,
            "paddingVertical": 16,
          },
          false,
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-end",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#004071",
                "fontFamily": "CircularPro-Bold",
                "fontSize": 17,
                "marginRight": 2,
              },
              false,
            ]
          }
        >
          3 hours
        </Text>
      </View>
      <View
        style={
          Object {
            "borderTopColor": "#979797",
            "borderTopWidth": 1,
            "marginTop": 16,
            "paddingTop": 16,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#004071",
              "fontFamily": "CircularPro-Bold",
              "fontSize": 15,
              "marginBottom": 16,
            }
          }
        >
          Testing PACKET Notification
        </Text>
        <Text
          style={
            Object {
              "color": "#58585a",
              "fontFamily": "CircularPro-Book",
              "fontSize": 13,
            }
          }
        >
          This is a testing packet level notification.
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#00a7e5",
            "borderBottomRightRadius": 5,
            "borderTopRightRadius": 5,
            "flex": 4,
            "justifyContent": "center",
            "paddingVertical": 12,
          },
          Object {
            "backgroundColor": "#54cbff",
          },
          false,
        ]
      }
    >
      <View
        accessibilityLabel="Purchase Pack 3 hours button"
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "width": 248.8,
            },
            Object {
              "alignItems": "stretch",
              "backgroundColor": "transparent",
              "borderColor": "#004071",
              "borderRadius": 32,
              "borderWidth": 1,
              "paddingVertical": 12,
            },
            Object {
              "paddingVertical": 8,
            },
            Object {
              "borderColor": "rgba(255, 255, 255, 0.7)",
            },
            false,
            false,
            false,
          ]
        }
        testID="Purchase Pack 3 hours button"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#004071",
                "fontFamily": "CircularPro-Bold",
                "fontSize": 20,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              Object {
                "fontSize": 15,
              },
              Object {
                "color": "#ffffff",
              },
              false,
              false,
              false,
            ]
          }
        >
          $5.00
        </Text>
      </View>
    </View>
  </View>
</View>
`;
